server/cluster/cluster.go

```diff
@@ -1021,30 +1021,34 @@ func (c *RaftCluster) RemoveStore(storeID uint64) error {
 	}
 
 	// Remove an offline store should be OK, nothing to do.
-	if store.IsOffline() {
+	if store.IsOffline() && store.IsPhysicallyDestroyed() == physicallyDestroyed {
 		return nil
 	}
 
 	if store.IsTombstone() {
 		return errs.ErrStoreTombstone.FastGenByArgs(storeID)
 	}
 
-	newStore := store.Clone(core.SetStoreState(metapb.StoreState_Offline))
+	if store.IsPhysicallyDestroyed() {
+		return errors.Errorf("The store is already physically destroyed")
+	}
```

Comment:
Maybe we can return `nil` here as the store is already `PhysicallyDestroyed`

Reply:
Since physically-destroyed is not allowed from true to false.

Reply:
I think we should define a errorCode like `ErrStoreTombstone` and to explain it as `The store have been force removed`